src/ioabc.cpp

```diff
@@ -347,6 +347,11 @@ void ABCInput::AddLayerElement()
             m_layer->AddChild(*iter);
         }
     }
+    // clean-up leftover data, if any
+    if (beam && !beam->GetChildCount()) delete beam;
```

Comment:
Sorry, but this is still cumbersome. If the beam a has a clef child but no note (hypothetical but not necessarily impossible) it would still not be deleted. I think you could set it to NULL above once it is added to m_layer or assigned to m_element, and then you can forget about the `!beam->GetChildCount()`. Or you do the same check as above `beam->FindDescendantByType(NOTE)`, but I think setting it to NULL would be clearer.